packages/react-dom/src/client/ReactDOMComponent.js

```diff
@@ -955,20 +895,32 @@ export function diffHydratedProperties(
       // satisfies our requirement. Our requirement is not to produce perfect
       // HTML and attributes. Ideally we should preserve structure but it's
       // ok not to if the visible content is still enough to indicate what
-      // even listeners these nodes might be wired up to.
+      // event listeners these nodes might be wired up to.
       // TODO: Warn if there is more than a single textNode as a child.
       // TODO: Should we use domElement.firstChild.nodeValue to compare?
       if (typeof nextProp === 'string') {
         if (domElement.textContent !== nextProp) {
-          if (__DEV__ && !suppressHydrationWarning) {
-            warnForTextDifference(domElement.textContent, nextProp);
+          if (__DEV__) {
+            didNotMatchHydratedChildrenPropValue(
+              tag,
+              rawProps,
+              (domElement: any),
```

Comment:
This `(domElement: any)` typecasting is ugly, but it was required to convince Flow that `Element` is compatible with `Instance` when typechecking with renderers other than `dom`, such as `yarn flow native`, where `react-dom` should not be used at all, but alas it's still typechecked.